src/components/TransactionsInterpreter/helpers/getTransactionDirection.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  TransactionDirection,
+  TransferTypeEnum,
+  UITransactionType
+} from './types';
+
+export function getTransactionDirection(
+  address: string,
+  transaction: UITransactionType,
+  receiver: string
+): TransactionDirection {
+  const directionOut = address === transaction.sender;
+  const directionIn = address === receiver;
+  const directionSelf = directionOut && directionIn;
+  const isScResult = transaction?.type === TransferTypeEnum.SmartContractResult;
+
+  let direction = TransactionDirection.OUT;
+  switch (true) {
+    case isScResult:
+      direction = TransactionDirection.INTERNAL;
+      break;
+    case directionSelf:
+      direction = TransactionDirection.SELF;
+      break;
+    case directionOut:
+      direction = TransactionDirection.OUT;
```

Comment:
somewhere above there is "Out" as string in another files, better use `TransactionDirection.OUT`